[Feature] 전기충전 주차장 필터링 기능 추가 및 api 연결

yejiis opened the pull request:
<!---- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->
<!---- Resolves: #(119) -->

## 💡 PR 유형
어떤 변경 사항이 있나요?
- [x] 새로운 기능 추가
- [ ] 버그 수정
- [x] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [x] 파일 혹은 폴더명 수정 및 삭제
- [x] 기타 사소한 변경

## ✅ PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.
- [x] 변경 사항에 대한 테스트를 지켰는가
- [x] 커밋 메시지가 팀 컨벤션을 지켰는가
- [x] 브랜치 명이 규칙에 맞는가
- [x] 기능 정상 동작 확인 완료

##  📝 작업 내용
<!-- 작업한 내용을 작성해주세요 -->
- 지도 페이지 컴포넌트 api 연결: 리스트, 모달, 마커
- 전기차 충전 필터링을 토글 버튼으로 관리. (url에 쿼리스트링을 추가하여 push)

yejiis committed:
feat: 주차장 마커 클릭 시 나타나는 모달 api 연결 및 useMap 코드 리팩토링 #118

yejiis committed:
feat: 전기차 필터링 기능 추가 및 api 연동 #119

yejiis committed:
feat: 상단 필터탭 버튼 수정 및 필터링 조건 추가 시 새로고침에서 router push로 변경 #119

yejiis committed:
Merge branch 'dev' of https://github.com/3-park-mate/FE-park-mate into feature/map-page-api

yejiis committed:
feat: 마커 클릭 시 나타나는 모달 api 연결 및 일부 수정
- api 연결
- clickmarker의 상태를 uuid로 관리하던 방식에서 marker의 info 데이터를 저장하고 있는 방식으로 수정(modal에 info를 전달하기 위함)
#119

yunchyn reviewed (commented):
1. map-service api가 따로 분리되어 있는데, 이게 백엔드 api의 parking-read-service에 해당되는 부분이라 parking-service 파일 안에 통합해도 좋지 않을까요?
2. ParkingLotSimpleInfoModal.tsx에서 훅 없이 데이터를 불러오고 있는데, 해당 파일에서 isLoading도 쓰고 있고, useFetchData 훅을 쓰면 좀 더 간단하지 않을까요? (안 썼다면 이유는 뭘까요?) 또 여기 코드를 아토믹 디자인으로 컴포넌트화 해도 될것 같습니다.
3. RatingOverview에 값이 없는 경우 0으로 초기화 하는 것보다 불러오는 쪽에서 조건 분기로 {RatingData && <RatingOverview ~/>} 해도 되지 않을까요? 아니면 RatingOverview 안에서 값이 없는 경우 return null 하거나 예를 들면 ParkingLotListCard.tsx에서 불러오고 있는데 이 쪽에서?
4. ParkingLotInfoType을 mapDataTypes에 추가로 작성해 주셨는데 이 타입이 이미 작성되어 있는 것 같습니다. parkingDataTypes.ts의 ParkingLotResponseDataType

yejiis committed:
refactor: 리뷰 반영 코드 및 파일 수정
- map-service를 parking-service에 통합
- ParkingLotSimpleinfoModal.tsx 에서 fetch훅을 사용하여 데이터를 불러오도록 변경
- RatingOverview에 값이 넘어오지 않는 경우에는 표시하지 않도록 변경
- 미사용 및 중복 type 선언 및 dummy data 제거
#122

yunchyn reviewed (approved):
확인했습니다
